### PR TITLE
Feature/pavlov artem/onboarding screen

### PR DESCRIPTION
**Pull request by ArtemPavlov72**
Add onboarding screen to project
- add framework WelcomeSheet
- add OnboardingScreenCoordinator
- add OnboardingModel
- add OnboardingService

### CODE REVIEW

File: Random/App/Sources/Navigation/MainScreenCoordinator/MainScreenCoordinator.swift [hunk 1]
```diff
@@ -511,6 +526,18 @@ private extension MainScreenCoordinator {
...
+    mainScreenModule?.returnModel { [weak self] model in
+      guard let self else {
+        return
+      }
+
+      if model.isFirstVisit {
+        openOnboardingScreen()
+      }
+    }
+  }
```
Comment (V1taS) [inline Random/App/Sources/Navigation/MainScreenCoordinator/MainScreenCoordinator.swift:540]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/V1taS/Random-Pro/blob/bec0b53c42473a446ca4096b7526472d0d52fecb/Random/App/Sources/Navigation/MainScreenCoordinator/MainScreenCoordinator.swift#L540"></span>Unused reference self in a capture list should be removed.<br />
<code>unused_capture_list</code> <code>MainScreenCoordinator.swift:540</code></td>
    </tr>
  </tbody>
</table>

File: Random/App/Sources/Navigation/MainScreenCoordinator/MainScreenCoordinator.swift [hunk 2]
```diff
@@ -520,6 +535,18 @@ private extension MainScreenCoordinator {
       window.overrideUserInterfaceStyle = isDarkTheme ? .dark : .light
     }
   }
+
+  func checkFirstVisit() {
+    mainScreenModule?.returnModel { [weak self] model in
```
Comment (V1taS) [inline Random/App/Sources/Navigation/MainScreenCoordinator/MainScreenCoordinator.swift:540]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/V1taS/Random-Pro/blob/12f4c2c5c3ef9f6d5f98b13f49f811564fac8fe6/Random/App/Sources/Navigation/MainScreenCoordinator/MainScreenCoordinator.swift#L540"></span>Unused reference self in a capture list should be removed.<br />
<code>unused_capture_list</code> <code>MainScreenCoordinator.swift:540</code></td>
    </tr>
  </tbody>
</table>

Comment (ArtemPavlov72) [reply Random/App/Sources/Navigation/MainScreenCoordinator/MainScreenCoordinator.swift]: Deleted IsFirstVisit from MainScreenModule and fix same code

File: Random/App/Tests/Service/ApplicationServicesMock.swift [hunk 3]
```diff
@@ -80,6 +80,11 @@ final class ApplicationServicesMock: ApplicationServices {
   var buttonCounterService: ButtonCounterService {
     buttonCounterServiceImpl
   }
+
+  var onboardingService: OnboardingService {
+    onboardingServiceIml
+  }
+
```
Comment (V1taS) [inline Random/App/Tests/Service/ApplicationServicesMock.swift:88]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/V1taS/Random-Pro/blob/4bc4f302303f7fae18ced6175c5ea23f3c8d9df6/Random/App/Tests/Service/ApplicationServicesMock.swift#L88"></span>Limit vertical whitespace to a single empty line. Currently 2.<br />
<code>vertical_whitespace</code> <code>ApplicationServicesMock.swift:88</code></td>
    </tr>
  </tbody>
</table>

File: Random/App/Tests/Service/OnboardingServce/OnboardingServiceMock.swift [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import WelcomeSheet
+import RandomNetwork
+@testable import Random
+
+final class OnboardingServiceMock: OnboardingService {
+  func saveWatchedStatus(to storage: StorageService, for models: [WelcomeSheet.WelcomeSheetPage]) {
+
+  }
+
+  func getOnboardingPagesForPresent(network: NetworkService, storage: StorageService, completion: (([WelcomeSheet.WelcomeSheetPage]) -> Void)?) {
```
Comment (V1taS) [inline Random/App/Tests/Service/OnboardingServce/OnboardingServiceMock.swift:20]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/V1taS/Random-Pro/blob/4bc4f302303f7fae18ced6175c5ea23f3c8d9df6/Random/App/Tests/Service/OnboardingServce/OnboardingServiceMock.swift#L20"></span>Line should be 140 characters or less: currently 145 characters<br />
<code>line_length</code> <code>OnboardingServiceMock.swift:20</code></td>
    </tr>
  </tbody>
</table>

File: Random/App/Tests/Service/OnboardingServce/OnboardingServiceMock.swift [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    
+  }
+
+
```
Comment (V1taS) [inline Random/App/Tests/Service/OnboardingServce/OnboardingServiceMock.swift:24]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/V1taS/Random-Pro/blob/4bc4f302303f7fae18ced6175c5ea23f3c8d9df6/Random/App/Tests/Service/OnboardingServce/OnboardingServiceMock.swift#L24"></span>Limit vertical whitespace to a single empty line. Currently 2.<br />
<code>vertical_whitespace</code> <code>OnboardingServiceMock.swift:24</code></td>
    </tr>
  </tbody>
</table>

File: Random/App/Tests/Service/ApplicationServicesMock.swift (hunk 3 above)
Comment (ArtemPavlov72) [reply Random/App/Tests/Service/ApplicationServicesMock.swift:88]: fixed

File: Random/App/Tests/Service/OnboardingServce/OnboardingServiceMock.swift (hunk 5 above)
Comment (ArtemPavlov72) [reply Random/App/Tests/Service/OnboardingServce/OnboardingServiceMock.swift:24]: fixed

File: Random/App/Tests/Service/OnboardingServce/OnboardingServiceMock.swift (hunk 4 above)
Comment (ArtemPavlov72) [reply Random/App/Tests/Service/OnboardingServce/OnboardingServiceMock.swift:20]: fixed

**Pull request merged**